src/backends/backend.js

```diff
@@ -41,23 +42,14 @@ class LocalStorageAuthStore {
   }
 }
 
-const slugFormatter = (template = "{{slug}}", entryData, slugConfig) => {
+const slugFormatter = (collection, entryData, slugConfig) => {
+  const template = collection.get('slug') || "{{slug}}";
   const date = new Date();
 
-  const getIdentifier = (entryData) => {
-    const validIdentifierFields = ["title", "path"];
-    const identifiers = validIdentifierFields.map((field) =>
-      entryData.find((_, key) => key.toLowerCase().trim() === field)
-    );
-
-    const identifier = identifiers.find(ident => ident !== undefined);
-
-    if (identifier === undefined) {
-      throw new Error("Collection must have a field name that is a valid entry identifier");
-    }
-
-    return identifier;
-  };
+  const identifier = entryData.get(selectIdentifier(collection));
+  if (identifier === undefined) {
```

Comment:
`if (identifier === undefined)` seems overly specific when `if (identifier)` seems to express the same semantics (none of `false`, `null`, and `0` are valid identifier field names either). If we'd rather avoid falsiness semantics, a better solution would be to use the second arguments to Immutable's `.get` and `.find` to explicitly return `false` when nothing is found.

Reply:
This was just copying existing code, but if we're going for 2.0 this should be fine.